src/ray/id.h

```diff
@@ -2,32 +2,57 @@
 #define RAY_ID_H_
 
 #include <inttypes.h>
+#include <limits.h>
 
+#include <chrono>
 #include <cstring>
+#include <mutex>
+#include <random>
 #include <string>
 
 #include "plasma/common.h"
 #include "ray/constants.h"
+#include "ray/util/logging.h"
 #include "ray/util/visibility.h"
 
 namespace ray {
 
-class RAY_EXPORT UniqueID {
+class DriverID;
+class UniqueID;
+
+std::mt19937 RandomlySeededMersenneTwister();
+
+template <typename T>
```

Comment:
Why not put the size in the template, like template <typename T, size_t SIZE>, we can specify the size when inherits the BaseID

Reply:
In the future, we may nest different IDs together. In this case, the template parameter size maybe not necessary.